File: 7_kyu/Make_a_function_that_does_arithmetic.lua
local kata = {}

kata.operations = {
    add = function(a,b) return a+b end,
    subtract = function(a,b) return a-b end,
    multiply = function(a,b) return a*b end,
    divide = function(a,b) return a/b end
}

function kata.arithmetic(a, b, operator)
    return kata.operations[operator](a,b)
end

return kata

File: 7_kyu/Make_a_function_that_does_arithmetic.go
package kata

func Arithmetic(a int, b int, operator string) int {
	var r int
	switch operator {
	case "add":
		r = a + b
	case "subtract":
		r = a - b
	case "multiply":
		r = a * b
	case "divide":
		r = a / b
	default:
		panic("invalid operator")
	}
	return r
}
